Gemfile

```diff
@@ -21,7 +21,14 @@ group :test do
   # but it's close enough to show the gempath handler can find a plugin
   # See test/unit/
   gem "train-test-fixture", path: "test/fixtures/plugins/train-test-fixture"
-  gem "mocha", "~> 1.1"
+  # Mocha ~1.1 is incompatible with minitest v5.19 and later
+  # More on the issue: https://github.com/freerange/mocha/issues/614
+  # It seems non-windows platforms lower than ruby 3.1 are affected
+  if !Gem.win_platform? && (Gem.ruby_version < Gem::Version.new("3.1"))
+    gem "mocha", "~> 1.1"
+  else
+    gem "mocha", "~> 2.1"
```

Comment:
Would it be blocking or breaking to just default to the `~> 2` mocha versions across the board now? Instead of keeping a unique subset in support of `mocha` < 2?

Reply:
Makes sense! Let me give it a try with a constraint of `~ 2` instead of `~ 2.1`. I introduced this condition because of the following error:

- On Ruby 2.7
   ```
   Could not find gem 'mocha (~> 2.1)' in any of the gem sources listed in your Gemfile.
   ```
- On Ruby 3.0 
   ```
   Could not find gem 'mocha (~> 2.1)' in locally installed gems.
   The source contains the following versions of 'mocha': 1.13.0, 1.14.0, 1.15.0, 1.16.0, 1.16.1, 2.0.1
   ```